src/mower_logic/src/mower_logic/behaviors/IdleBehavior.cpp

```diff
@@ -70,8 +70,9 @@ Behavior *IdleBehavior::execute() {
         const auto last_config = getConfig();
         const auto last_status = getStatus();
         if (manual_start_mowing ||
-            (last_config.automatic_start && (last_status.v_battery > last_config.battery_full_voltage && last_status.mow_esc_status.temperature_motor < last_config.motor_cold_temperature &&
-             !last_config.manual_pause_mowing))) {
+            (last_config.automatic_mode == eAutoMode::AUTO || (last_config.automatic_mode == eAutoMode::SEMIAUTO && last_config.active_semiautomatic_task == true))
```

Comment:
active_semiautomatic_task only gets set to true, never to false. How does it stop after one full rotation?

Reply:
I have already fixed that issue on my local branch. The semi-automatic task should be set to false whenever we call MowingBehavior::reset()